### PR TITLE
Update train compare script

### PR DESCRIPTION
**Pull request by Ghostvv**
**Proposed changes**:
- Update train compare script to allow whole config comparison

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

**Comment by Ghostvv** [comment]
Should I increase version to `1.3`?

**Comment by Ghostvv** [comment]
There is nothing really to migrate, just retrain

File: CHANGELOG.rst [hunk 1]
```diff
@@ -6,12 +6,12 @@ Rasa Change Log
 All notable changes to this project will be documented in this file.
 This project adheres to `Semantic Versioning`_ starting with version 1.0.
 
-[Unreleased 1.2.3] - `master`_
+[Unreleased 1.3] - `master`_
```
Comment (tabergma) [inline CHANGELOG.rst]: You should create a new section for the release (eg by moving the items from the collected master section) and create a new master logging section

File: docs/migration-guide.rst [hunk 2]
```diff
@@ -11,6 +11,25 @@ Migration Guide
...
+  It is not possible to load previously trained models. Please make sure to retrain a
+  model before trying to use it with this improved version.
+
+General
+~~~~~~~
+
+- **Compare** mode of ``rasa train core`` allows the whole core config comparison.
+  Therefore, we changed the naming of trained models. They are named by config file
+  name instead of policy name. Old naming style will not be read correctly when
+  creating **compare** plots. Please remove old trained models in comparison folder
```
Comment (tabergma) [inline docs/migration-guide.rst]:
```suggestion 
  creating **compare** plots (`rasa test core`). Please remove old trained models in comparison folder
```

**Pull request merged**